property.go

```diff
@@ -65,13 +69,41 @@ func ExtractProperty(err error, key Property) (interface{}, bool) {
 }
 
 var (
-	propertyContext = RegisterProperty("ctx")
-	propertyPayload = RegisterProperty("payload")
+	propertyContext    = RegisterProperty("ctx")
+	propertyPayload    = RegisterProperty("payload")
+	propertyUnderlying = RegisterProperty("underlying")
 )
 
 func newProperty(label string) Property {
-	return Property{
-		id:    nextInternalID(),
-		label: label,
+	p := Property{
+		&property{
+			id:    nextInternalID(),
+			label: label,
+		},
+	}
+	return p
+}
+
+// propertyMap represents map of properties.
+// Compared to builtin type, it uses less allocations and reallocations on copy.
+// It is simple binary search tree that relies on property id randomness for balancing.
```

Comment:
This comment is false. Please revisit all the new comments and see if they are up to date.

Reply:
Ok. Forgot about.